File: 2002B2/VB/WinForms-Inheritance-Vehicle/Vehicle/VehicleBO/Boat.vb

Namespace VehicleDemo
    'For boats we need to override the implementation of the 
    'ComputeRange function to extend the range of the vehicle
    Public Class Boat
        Inherits Vehicle

        'We need to override the ComputeRange function to add support for boats
        'For boats we want to increase the couputed range.  We will call the
        'ComputeRange function on MyBase which is the Vehicle class
        Overrides Function ComputeRange() As Integer
            'If this is a sailboat then we double the calculated range
            ComputeRange = MyBase.ComputeRange * 2
        End Function

    End Class
    
End Namespace

File: 2002B2/CS/WinForms-Inheritance-Vehicle/Vehicle/VehicleSampleBO/Boat.cs
using System;

namespace VehicleDemo
{
	//For boats we need to override the implementation of the ComputeRange function
	
	/// <summary>
	/// Summary description for Boat.
	/// </summary>
	public class Boat : Vehicle
	{
		//We need to override the ComputeRange function to add support for boats
		//For boats we want to double the couputed range.  We will call the
		//ComputeRange function on MyBase which is the Vehicle class
		public override int ComputeRange()
		{
			//This is a boat so we double the calculated range
			return base.ComputeRange() * 2;
		}
			public Boat()
			{
				//
				// TODO: Add constructor logic here
				//
			}
	}
}
